Content.Server/GameObjects/Components/Items/FloorTileItemComponent.cs

```diff
@@ -89,12 +85,12 @@ public void AfterInteract(AfterInteractEventArgs eventArgs)
                     }
                 }
             }
-            else if(HasBaseTurf(desiredTile, "space"))
+            else if(HasBaseTurf((ContentTileDefinition)_tileDefinitionManager[_outputTiles[0]], "space"))
             {
                 mapGrid = _mapManager.CreateGrid(locationMap.MapId);
                 mapGrid.WorldPosition = locationMap.Position;
                 location = new EntityCoordinates(mapGrid.GridEntityId, Vector2.Zero);
-                PlaceAt(mapGrid, location, desiredTile.TileId, mapGrid.TileSize/2f);
+                PlaceAt(mapGrid, location, _tileDefinitionManager[_outputTiles[0]].TileId, mapGrid.TileSize/2f);
```

Comment:
You're not doing this correctly... Both this else if and the if above should be part of the for loop for all output tiles, and break when you correctly place a tile.